t/Util.pm

```diff
@@ -820,4 +824,65 @@ sub spawn_dns_server {
     return $server;
 }
 
+sub run_openssl_client {
+    my($opts) = @_;
+    my $port = $opts->{port} or croak("`port` is required!");
+    my $san = $opts->{san};
+    my $host = $opts->{host} // '127.0.0.1';
+    my $path = $opts->{path} // '/';
+    my $timeout = $opts->{timeout} // 2.0;
+    my $request = $opts->{request};
+    my $request_default = $opts->{request_default} // 0;
+    my $ossl_opts = $opts->{opts} // '';
+    my $ossl_cmd = $opts->{ossl_cmd} // 'openssl';
+
+    my $tempdir = tempdir();
+    my $cmd = "$ossl_cmd s_client $ossl_opts -connect $host:$port";
+    if (defined $san && $san ne '') {
+        $cmd = $cmd." -servername $san";
+    }
+    diag("run_openssl_client: $cmd");
+    my $cpid = open3(my $chld_in, my $chld_out, my $chld_err = gensym, $cmd);
```

Comment:
Just a remainder that IPC::Open3 is sensitive to deadlocks. As it uses pipe in both directions, we might end up seeing the sub-command blocked on write, while us being blocked on write. I would assume that the code below would block for example when the server is a simple echo server and the we write lots of data.

The way to address this problem is to redirect stdout and stderr of the sub-command to files in temporary directory, then read them once the command execution completes.

I would not mind using IPC::Open3 for the time being, but pointing that out as I see tempdir above.

Reply:
I agree.  Since we don't have such test scenarios, let's keep it as is?